src/arch/arm/32/kernel/vspace.c

```diff
@@ -2608,7 +2608,7 @@ exception_t decodeARMMMUInvocation(word_t invLabel, word_t length, cptr_t cptr,
 
         if (unlikely(cap_get_capType(untyped) != cap_untyped_cap ||
                      cap_untyped_cap_get_capBlockSize(untyped) !=
-                     seL4_ASIDPoolBits) || cap_untyped_cap_get_capIsDevice(untyped)) {
+                     seL4_ASIDPoolBits || cap_untyped_cap_get_capIsDevice(untyped))) {
```

Comment:
Just curious, did you observe differences in the assembly code before and after the change? @mbrcknl

Reply:
I didn't think to look until you asked. :-)

There are differences in the assembly, but it's such a huge function, I can't really figure out what the differences mean. With this change, the assembly is 24 bytes longer, and that means offsets to things after the function are different. That makes it quite difficult to diff.

Reply:
Thanks. Agreed that the if conditions are quite complex.